lightningd/peer_control.c

```diff
@@ -690,6 +702,13 @@ static void gossipd_getpeers_complete(struct subd *gossip, const u8 *msg,
 		/* Fake state. */
 		json_add_string(response, "state", "GOSSIPING");
 		json_add_pubkey(response, "id", ids+i);
+		for (size_t j = 0; j < tal_count(nodes); j++) {
+			if (nodes[j]->addresses != NULL && pubkey_eq(&nodes[j]->nodeid, ids+i)) {
+				json_add_string_escape(response, "alias", (char*)nodes[j]->alias);
+				json_add_hex(response, "color", nodes[j]->color, ARRAY_SIZE(nodes[j]->color));
+				break;
+			}
+		}
```

Comment:
These two loops are both identical, we could extract it into a "maybe_add_alias_color(response, ids+i, nodes)".  It's O(N^2) which is probably OK.